### DISCUSSION TITLE
Question about VBCSCompiler.dll process lifetime

### DISCUSSION THREAD

**Discussion by am11** [dotnet/msbuild#11010] [Q&A] [answered]
I was doing some investigation of issue discussed in https://github.com/dotnet/sdk/pull/44922 and found the following comment:

https://github.com/dotnet/msbuild/blob/9c89563239bd60739920991211649d899b32ecb4/src/Build/BackEnd/Components/Communications/DetouredNodeLauncher.cs#L131-L133

but in reality, `VBCSCompiler.dll` process keeps running after executing `dotnet msbuild` (or `dotnet build`), we get:

```sh
$ rm -r obj bin # evict cache
$ dotnet build
$ ps -e -o pid,ppid,args | grep -i vbcs | fold -65
71304     1 /Users/am11/.dotnet10/dotnet exec /Users/am11/.dotn
et10/sdk/10.0.100-alpha.1.24567.3/Roslyn/bincore/VBCSCompiler.dll
 -pipename:_UckHksr8RUqDGzqnwgGw7cOMBqfRYBBs8oEnOvtDzY
```

It does go away on its own after exactly 600 seconds (10 minutes). If this is by design:
1. where is this timeout specified?
2. is there any user-facing configuration to change it (make it zero)?

**Comment by KalleOlaviNiemitalo** [comment, marked as answer]
Perhaps `dotnet build --disable-build-servers`; I did not test.

Also, `dotnet build-server shutdown --vbcscompiler`; but see <https://github.com/dotnet/sdk/issues/20183>.

The 10-minute timeout seems to be at <https://github.com/dotnet/roslyn/blob/9bea39d29a00719039d0d3705a15d384bdc385da/src/Compilers/Server/VBCSCompiler/ServerDispatcher.cs#L43-L46>.  It would be configurable in appsettings <https://github.com/dotnet/roslyn/blob/9bea39d29a00719039d0d3705a15d384bdc385da/src/Compilers/Server/VBCSCompiler/BuildServerController.cs#L64> but the configuration is only read on .NET Framework <https://github.com/dotnet/roslyn/blob/9bea39d29a00719039d0d3705a15d384bdc385da/src/Compilers/Server/VBCSCompiler/VBCSCompiler.cs#L25-L30>.

**Reply by am11** [reply]
Nice! I knew about build-server shutdown, but `--disable-build-servers` (or `-p:UseSharedCompilation=false`) does not spawn it (or shutdown the server after build completion). 👍
